File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        s = re.sub(r'[^a-zA-Z0-9]', '', s)
        s = s.lower()
        if not s:
            return True
        left = 0
        right = len(s) - 1
        while left < right:
            if s[left] != s[right]:
                return False
            left += 1
            right -= 1
        return True
        

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String ls = s.toLowerCase();
        String alphanumeric = "abcdefghijklmnopqrstuvwxyz0123456789";
        StringBuilder ns = new StringBuilder();

        // Only keep alphanumeric characters
        for (int i = 0; i < ls.length(); i++) {
            if (alphanumeric.indexOf(ls.charAt(i)) != -1) {
                ns.append(ls.charAt(i));
            }
        }

        // Check if the cleaned string is a palindrome
        String cleaned = ns.toString();
        String reversed = ns.reverse().toString();
        return cleaned.equals(reversed);

    }
}